### ISSUE TITLE
RFE: bash: 

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#1133]
#### For new checks and feature suggestions

- [?] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
  - No idea as it requires JavaScript
  - No warnings from shellcheck 0.4.6-1 in Debian
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet that shows the problem:

```sh

#!/bin/bash
alias foo='echo bar'
foo
```

#### Here is an illustration of the results:

```
$ dash foo.sh
bar
$ bash foo.sh
foo.sh: line 3: foo: command not found
$ bash --posix foo.sh
bar
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

line 3: bash does not support using an alias when not in interactive mode and `shopt expand_aliases` is not set.

#### Reference

https://www.gnu.org/software/bash/manual/html_node/Aliases.html

**Comment by Dmole** [comment]
shellcheck.net currently gives no useful warnings about this.

**Comment by koalaman** [comment]
Is there a good way to warn that won't also trigger for `shellcheck -s bash ~/.bashrc`?
